frontend/src/assets/js/main.js

```diff
@@ -76,38 +82,43 @@ function connect(username, roomCode, difficulty, hackyObject) {
         elem.id = `collaborator-${message.data.username}`;
         elem.appendChild(document.createTextNode(message.data.username));
         list.appendChild(elem);
+        break;
 
       case "disconnect":
         collaborators = collaborators.filter((collaborator) => {
-          collaborator.id !== message.data.id;
+          return collaborator.id !== message.data.id;
         });
         var list = document.getElementById("collabul");
         var elem = document.getElementById(
```

Comment:
Commenting here because I can't comment on line 81. Also move `var elem = ...` to above the switch statement and declare this one as `let`. It is only used in the connect and disconnect so they shouldn't interfere with each other ( I think)

Reply:
However, we can actually get the element only when it's created (basically when the connect event is sent).

Reply:
Guess this makes sense within it's scope as you don't want to be creating the list element on each request